ui/plugins/controller.js

```diff
@@ -123,7 +130,11 @@ treeherder.controller('PluginCtrl', [
                     $scope.jobSearchSignature = $scope.job.signature;
                     $scope.jobSearchStrHref = getJobSearchStrHref($scope.jobSearchStr);
                     $scope.jobSearchSignatureHref = getJobSearchStrHref($scope.job.signature);
-
+                    var generateUrlForPerf = function(projectName, jobId, signature, resultSetId) {
```

Comment:
This function is only called once, so I don't think it's necessary. Just put the code inline where it's used.